cpp/CMakeLists.txt

```diff
@@ -364,7 +385,7 @@ Imported Targets:
 set(code_string
 [=[
 
-if(NOT TARGET raft::Thrust)
+if(RAFT_ENABLE_thrust_DEPENDENCY AND NOT TARGET raft::Thrust)
```

Comment:
The variable `RAFT_ENABLE_thrust_DEPENDENCY` doesn't exist when users of raft call `find_package(raft)`.

Instead you should conditionally append this logic to `code_string` instead

Reply:
Something like:
```cmake
set(code_string )
if(RAFT_ENABLE_thrust_DEPENDENCY)
  string(APPEND code_string
  [=[
  if(NOT TARGET raft::Thrust)
    thrust_create_target(raft::Thrust FROM_OPTIONS)
  endif()
  ]=])
endif()

string(APPEND code_string
[=[
...
```